src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -38,13 +38,13 @@ public bool AllowUnsafeBlocks
         public string ApplicationConfiguration
         {
             set { _store[nameof(ApplicationConfiguration)] = value; }
-            get { return (string)_store[nameof(ApplicationConfiguration)]; }
+            get { return _store[nameof(ApplicationConfiguration)] as string; }
```

Comment:
I find it to be more readable (especially when the return type is already specified). I can revert these particular changes if you find it to be otherwise.